# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/push/PushSubscription.cpp
# Commit: 7802e5f1bd13
# Full Hash: 7802e5f1bd1395f19f2d78636787c41f64192bd7
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-04-19 15:22:44
# Regressor Bug: 1769913
# File Overlap Count: 3
# Description:
#   Bug 1769913 - P3 Remove WorkerRunnable::mWorkerPrivate. r=dom-worker-reviewers,asuth
#   
#   WorkerRunnable no longer keeps a raw pointer(mWorkerPrivate) for the associated WorkerPrivate in this patch.
#   Removing the WorkerRunnable::mWorkerPrivate needs to fix the following problems.
#   
# ==============================================================================

diff -r ffc69aa8f2c3 -r 7802e5f1bd13 dom/push/PushSubscription.cpp
--- a/dom/push/PushSubscription.cpp	Fri Apr 19 09:41:57 2024 +0000
+++ b/dom/push/PushSubscription.cpp	Fri Apr 19 09:41:58 2024 +0000
@@ -59,7 +59,7 @@
   UnsubscribeResultRunnable(WorkerPrivate* aWorkerPrivate,
                             RefPtr<PromiseWorkerProxy>&& aProxy,
                             nsresult aStatus, bool aSuccess)
-      : WorkerThreadRunnable(aWorkerPrivate, "UnsubscribeResultRunnable"),
+      : WorkerThreadRunnable("UnsubscribeResultRunnable"),
         mProxy(std::move(aProxy)),
         mStatus(aStatus),
         mSuccess(aSuccess) {
@@ -117,7 +117,7 @@
     WorkerPrivate* worker = mProxy->GetWorkerPrivate();
     RefPtr<UnsubscribeResultRunnable> r = new UnsubscribeResultRunnable(
         worker, std::move(mProxy), aStatus, aSuccess);
-    MOZ_ALWAYS_TRUE(r->Dispatch());
+    MOZ_ALWAYS_TRUE(r->Dispatch(worker));
 
     return NS_OK;
   }